server/middleware.go

```diff
@@ -21,24 +45,27 @@ func withMetrics(
 	return func(w http.ResponseWriter, r *http.Request) error {
 		recordDur := m.RecordRPCServerRequest(r.Method)
 
-		err := handleFn(w, r)
+		scw := newStatusCaptureWriter(w)
+		err := handleFn(scw, r)
 		if err != nil {
+			commitMode := "unknown"
+			certVersion := "unknown"
 			var metaErr MetaError
 			if errors.As(err, &metaErr) {
-				recordDur(w.Header().Get("status"), string(metaErr.Meta.Mode), string(metaErr.Meta.CertVersion))
-			} else {
-				recordDur(w.Header().Get("status"), string("unknown"), string("unknown"))
+				commitMode = string(metaErr.Meta.Mode)
+				certVersion = string(metaErr.Meta.CertVersion)
 			}
+			recordDur(strconv.Itoa(scw.status), commitMode, certVersion)
 			return err
 		}
-		// TODO: some routes don't have a version byte, such as /put simple commitments
 		versionByte, err := parseVersionByte(w, r)
 		if err != nil {
-			// we assume that every route will set the status header
-			recordDur(w.Header().Get("status"), string(mode), string(versionByte))
-			return fmt.Errorf("metrics middleware: error parsing version byte: %w", err)
+			// Some routes don't have a version byte, such as /put simple commitments.
```

Comment:
why would /put simple commitments not have a version byte but OP /put would?

Reply:
Good catch.... I was totally wrong here. I updated to: [096003c](https://github.com/Layr-Labs/eigenda-proxy/pull/200/commits/096003c396b2fa32840616452ced021d00109d75)
Realized our version bytes are hardcoded for all the PUT routes.. bit ugly how I did some of the routing now I'm realizing. But let me know what you think of these changes. You'll have to change this when you get the v2 changes in.

Reply:
Oh also for reference it seems like its the keccak commitments that don't have a version byte: https://github.com/Layr-Labs/eigenda-proxy/blob/096003c396b2fa32840616452ced021d00109d75/server/routing.go#L66